src/m365/spo/commands/site/site-admin-add.ts

```diff
@@ -0,0 +1,358 @@
+import { Logger } from '../../../../cli/Logger.js';
+import config from '../../../../config.js';
+import GlobalOptions from '../../../../GlobalOptions.js';
+import request, { CliRequestOptions } from '../../../../request.js';
+import { entraGroup } from '../../../../utils/entraGroup.js';
+import { formatting } from '../../../../utils/formatting.js';
+import { FormDigestInfo, spo } from '../../../../utils/spo.js';
+import { validation } from '../../../../utils/validation.js';
+import SpoCommand from '../../../base/SpoCommand.js';
+import commands from '../../commands.js';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface AdminUserResult {
+  email: string;
+  loginName: string;
+  name: string;
+  userPrincipalName: string;
+}
+
+interface AdminResult {
+  value: AdminUserResult[];
+}
+
+interface IGraphUser {
+  userPrincipalName: string;
+}
+
+interface ISiteOwner {
+  LoginName: string;
+}
+
+interface ISPSite {
+  Id: string;
+}
+
+interface ISiteUser {
+  Id: number;
+}
+
+interface Options extends GlobalOptions {
+  siteUrl: string;
+  userId?: string;
+  userName?: string;
+  groupId?: string;
+  groupName?: string;
+  primary?: boolean;
+  asAdmin?: boolean;
+}
+
+class SpoSiteAdminAddCommand extends SpoCommand {
+  public get name(): string {
+    return commands.SITE_ADMIN_ADD;
+  }
+
+  public get description(): string {
+    return 'Adds a user or group as a site collection administrator';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+    this.#initTypes();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        siteUrl: typeof args.options.siteUrl !== 'undefined',
```

Comment:
As this is a required option, this can be removed

Reply:
ok, I have also removed it from site admin list